cmake/Version.cmake

```diff
@@ -22,9 +22,10 @@ macro(get_version NAME VERSION_VAR)
     elseif(DEFINED OMNIGRES_VERSION)
         set("${VERSION_VAR}" "${OMNIGRES_VERSION}")
     else()
-        execute_process(COMMAND git describe  --tags --always --match "^v"
-                WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+        execute_process(COMMAND bash -c "./get-extension-version.sh versions.txt ${_name}"
```

Comment:
Is there anything that's preventing us from doing an equivalent reader in CMake here? At least that was we're not jumping between disconnected components.

Something to this tune:

```cmake
file(STRINGS versions.txt lines)
foreach(line ${lines})
# ...
```

Reply:
Also, referring to `versions.txt` without a path may backfire. Perhaps a location relative to Version.cmake (that is, the current list dir) can be better? 

What if we save `CMAKE_CURRENT_LIST_DIR` before we define the macro in a globally cached variable and then use it to`${dir}/../version.txt` to find the file?

Reply:
> Is there anything that's preventing us from doing an equivalent reader in CMake

The bash script detects duplicate entries of an extension as well, errors out if that's the case

Reply:
I suspect this might not work as expected when extensions are built individually, as their `CMAKE_SOURCE_DIR` won't be the root.

Reply:
The PR did pass [build-extensions-independently](https://github.com/omnigres/omnigres/actions/runs/8096610533/job/22125731018?pr=504#logs)

Reply:
Removed the bash script and made the changes you suggested